.github/bin/download-maven-dependencies.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Enable all known profiles to make sure their plugin dependencies are downloaded
+# GIB should be disabled even though it's profile is activated, to make sure it doesn't skip any submodules
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable dependency:go-offline
```

Comment:
This disables the active-by-default profiles, so not all profiles are effectively active.

In fact, profiles can have exclusionary configuration, so for example compilation with `errorprone-compiler` likely needs *different* (not just more) dependencies than compllation wiithout it.

Therefore I think we should invoke go-offline twice, with `errorprone-compiler` and without it.
not sure about `ci` profile (do we need to invoke with it? why? let's have a comment)

Reply:
It turns out the `ci` profile doesn't do anything w.r.t. Maven dependencies, so I'll remove it here.
The only profiles which add their own dependencies are `errorprone-compiler` and `gib`.
Plugins also have their own profiles to control which tests get executed by Surefire, but I don't think that changes the dependency list.